internal/topology/upgrade/clusterctl_upgrade_test.go

```diff
@@ -257,6 +281,18 @@ func createT1ClusterClass(g *WithT, ns *corev1.Namespace, ct1 client.Client) *cl
 			Namespace: ns.Name,
 			Name:      "test-infra-cluster-template-v1beta1-t1",
 		},
+		Spec: testt1v1beta1.TestResourceTemplateSpec{
+			Template: testt1v1beta1.TestResourceTemplateResource{
+				Spec: testt1v1beta1.TestResourceSpec{
+					BoolToPtrBool:     true,
+					PtrStringToString: ptr.To("Something"),
+					Int32ToPtrInt32:   int32(4),
+					StructWithOnlyOptionalFields: testt1v1beta1.StructWithOnlyOptionalFields{
+						A: "Something",
+					},
```

Comment:
Do these values give us the actual coverage?

Reply:
considering this templates and the other ones, yes